src/include/migraphx/allocation_model.hpp

```diff
@@ -267,7 +267,7 @@ struct allocation_model
     private_detail_te_handle_base_type& private_detail_te_get_handle()
     {
         assert(private_detail_te_handle_mem_var != nullptr);
-        if(not private_detail_te_handle_mem_var.unique())
+        if(private_detail_te_handle_mem_var.use_count() != 1)
```

Comment:
shoulde be >= 1 otherwise  use_count() == 0 is valid here to clone

Reply:
done. I updated in all places and the te.py script, too